packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -366,21 +366,35 @@ export function TransferPanelMain({
       l1: null,
       l2: null
     }
+    const { bridgeTokens } = arbTokenBridge
 
-    if (!selectedToken) {
+    if (!selectedToken || !bridgeTokens) {
       return result
     }
 
     if (erc20L1Balances) {
       result.l1 = erc20L1Balances[selectedToken.address] ?? null
     }
 
-    if (erc20L2Balances && selectedToken.l2Address) {
-      result.l2 = erc20L2Balances[selectedToken.l2Address] ?? null
+    const erc20L2Address =
+      bridgeTokens[selectedToken.address.toLowerCase()]?.l2Address
+
+    if (erc20L2Balances) {
+      if (erc20L2Address) {
+        result.l2 = erc20L2Balances[erc20L2Address] ?? null
+      } else {
+        result.l2 = constants.Zero
+      }
     }
 
     return result
-  }, [erc20L1Balances, erc20L2Balances, selectedToken])
+  }, [
+    erc20L1Balances,
+    erc20L2Balances,
+    selectedToken,
+    arbTokenBridge,
+    destinationAddressOrWalletAddress
```

Comment:
why is `destinationAddressOrWalletAddress` added?

Reply:
After adding `arbTokenBridge` I've noticed a bit of inconsistent balances behavior when changing the destination address. 

Adding `destinationAddressOrWalletAddress` to dependencies fixes that, and besides it just should be part of it anyway.